java/semantickernel-api/src/main/java/com/microsoft/semantickernel/hooks/FunctionInvokedEventArgs.java

```diff
@@ -0,0 +1,31 @@
+package com.microsoft.semantickernel.hooks;
+
+import com.microsoft.semantickernel.orchestration.FunctionResult;
+import com.microsoft.semantickernel.orchestration.KernelFunction;
+import com.microsoft.semantickernel.orchestration.contextvariables.KernelArguments;
+
+public class FunctionInvokedEventArgs<T> implements HookEvent {
```

Comment:
It seems odd to me that FunctionInvokedEventArgs would implement HookEvent. I'd expect FunctionInvokedEventArgs to be called FunctionInvokedEvent since it is-a HookEvent. It would be clear (to me, at least) if you renamed HookEvent to HookArgs and s/EventArgs/Args/. Then you'd have FunctionInvokedArgs implements HookArgs, which makes better sense (to me, at least) that these are arguments to/from a function invocation and in use you get `FunctionInvokingHook preHook = functionInvokingArgs -> blah`